st2common/st2common/models/db/execution.py

```diff
@@ -141,11 +152,27 @@ def mask_secrets(self, value):
                         p: "string" for p in liveaction["parameters"]["response"]
                     },
                 )
-
+        
+        output_schema_result = ActionExecutionDB.result.parse_field_value(
+            result["result"]
+        )
+        
+        # accessing parameters marked secret as true in the output_schema
+        # and masking them for the result in action execution API
+        if output_schema_result:
```

Comment:
I suggest moving this block into a function called `mask_secret_output` and put it under the module https://github.com/StackStorm/st2/blob/master/st2common/st2common/util/output_schema.py. This will help whoever is making changes next can look at the output_schema functionalities in one place.